cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -508,6 +509,38 @@ pub trait ClientTrait {
     ) -> Result<HashMap<String, Vec<StoreInfo>>, CatalogClientError>;
 }
 
+/// Wrapper around `semver::VersionReq` which attempts to normalize semver
+/// strings in a format that the catalog supports.
+#[derive(Debug, Error)]
+struct SemverNormalized(VersionReq);
+
+impl TryFrom<String> for SemverNormalized {
+    type Error = semver::Error;
+
+    fn try_from(s: String) -> Result<Self, Self::Error> {
+        let version = VersionReq::parse(&s)?;
+        Ok(SemverNormalized(version))
+    }
+}
+
+// Forked from:
+// https://github.com/dtolnay/semver/blob/1.0.26/src/display.rs#L33-L46
+impl std::fmt::Display for SemverNormalized {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.0.comparators.is_empty() {
+            return f.write_str("*");
+        }
+        for (i, comparator) in self.0.comparators.iter().enumerate() {
+            if i > 0 {
+                // This differs from upstream by omitting spaces in the separators.
+                f.write_str(",")?;
+            }
+            write!(f, "{}", comparator)?;
+        }
+        Ok(())
+    }
+}
+
 impl ClientTrait for CatalogClient {
     /// Wrapper around the autogenerated
     /// [catalog_api_v1::Client::resolve_api_v1_catalog_resolve_post]
```

Comment:
suggestion: amend the comment that we _try_ to normalize semver requirements but eventually fall back to the original if we're unable to.